Create ISSUE_TRIAGE.md (#75377)
CleverRaven/Cataclysm-DDA: doc/ISSUE_TRIAGE.md

## Why triage?

Triage is a process of sanity checking and prioritizing issues. The term is adopted from medical contexts where patients are briefly examined to decide whether they need immediate care, or if their issue can wait until patients at greater risk can be handled first.

> Software issue triage is not life-or-death, but it does adress the same fundamental problem, which is that thre is a limited amount of attention and skill that can be broght to bear on issues at any given time, and some issues have higher *impact* than others.

Reviewer: ```suggestion
Software issue triage is not life-or-death, but it does adress the same fundamental problem, which is that there is a limited amount of attention and skill that can be brought to bear on issues at any given time, and some issues have higher *impact* than others.
```

If issues are properly triaged, contributors with time available can focus on getting things done instead of searching through issues for something to work on and fix.

## How does triage work?
Look at new issues or revisit old issues after a while and apply various labels:
1. Characterize as a bug report or a feature suggestion
2. if bug report:
    1. label with priority
    2. check for validity
3. if feature suggestion:
    1. evaluate for validity

> 4. in either case, also apply labels indicating what areas of the game it immpacts, i.e. content/json vs features/code, items vs monsters vs translation etc.

Reviewer 2: ```suggestion
4. in either case, also apply labels indicating what areas of the game it impacts, i.e. content/json vs features/code, items vs monsters vs translation etc.
```

### Is this a bug report or a feature request?
Sometimes this is simple, is the game crashing, is it losing save data? If so it's definitely a bug.
Likewise, new content or new ways to interact with the game are always feature requests.

> Other times this is a surprisingly tricky question. "the game isn't working like I expect it to" doesn't actually mean the game is doing anything wrong, and even if it is (i.e. it sets up an expectation but doesn't meet it), the problem might be where the expectation is set rather than the point where the expectation is (not) met.

Reviewer 2: ```suggestion
Other times this is a surprisingly tricky question. "The game isn't working like I expect it to" doesn't actually mean the game is doing anything wrong, and even if it is (i.e. it sets up an expectation but doesn't meet it), the problem might be where the expectation is set rather than the point where the expectation is (not) met.
```

At the end of the day, this is a judgement call about what the triager understands the game to be designed to do vs what it's actually doing. If the game's behavior is consistent with the design of the game, we'd call that a request to change it a feature request, if the game is not acting as designed, then it's a bug.

### Bug priority
Roughly our priority ordering is:
1. crashes
2. loss of data. This includes save migration from supported stable versions, but not ancient stable versions or experimental versions. To be clear migration *to* stable versions from experimental versions *is* included. This also includes chunks of map being lost or overwritten.
3. character death or item loss due to bugs (this is unfortunately not itself a bug but the outcome of a bug co it can be very subjective)
4. blatant "lack of consistency" bugs (characters, items or monsters having stats an order of magnitude or so off from reality, mechanics that make no sense).

> 5. Poor UX due to e.g. excessive keypresses for an activity, tricky or misleading input, poor wording of UI elements. (maybe actually a feature request)
> 6. minor inconsistencies e.g. things that are only apparent by comparing json values, typos, 

Reviewer 2: ```suggestion
5. poor UX due to e.g. excessive keypresses for an activity, tricky or misleading input, poor wording of UI elements. (maybe actually a feature request)
6. minor inconsistencies e.g. things that are only apparent by comparing json values, typos, and so on.
```

> This is a bit more granular than necessary, 1, 2 and 3 are critical, 4 and 5 are important and everything else is null priority.

Reviewer: There is no 3.

You CAN add those details yourself if you are able to and so inclined, but it's not a necessary part of the process.
If an issue does not have these elements, and especially if no one else is able to reproduce the issue other than the reporter, that issue is unfortunately just noise, and the eventual outcome is closing the issue until someone can provide those details.

### Feature request validity
Feature requests should be specific, achievable and relevant (some of you might be thinking "SMART goals" and you'd be right, but Measurable and Time-bound aren't applicable).  
- Specific means the issue should clearly describe what the feature should do or what content should be added.  
- Achievable both means the feature should be not just *possible*, but also *feasible*. "Use generative AI for NPC conversations" fails both. "Use detailed nutrition information to make food better" is possible but would require a prohibitive amount of work.  

> - Relevant means that the feature/content should *improve the game*, there's a certain amount of "add more stuff" that's self-justifying under the vague rationale of "make the world more believable", but taken to an extreme this is an unsustainable thing to do. Likewise features for the sake of features that don't actually impact the game in any way should be discouraged, or at least scoped to match the effort required to implement them with thebenefit they provide.

Reviewer 2: ```suggestion
- Relevant means that the feature/content should *improve the game*, there's a certain amount of "add more stuff" that's self-justifying under the vague rationale of "make the world more believable", but taken to an extreme this is an unsustainable thing to do. Likewise features for the sake of features that don't actually impact the game in any way should be discouraged, or at least scoped to match the effort required to implement them with the benefit they provide.
```
